tests/presets.spec.js

```diff
@@ -740,6 +740,445 @@ describe('Presets', function() {
+            m.chai.expect(error).to.be.an.instanceof(Error);
+            m.chai.expect(error.code).to.equal('ENOENT');
+            done();
+          });
+        });
+
+      });
+
+      describe('updating Cargo.toml with unchanged version', function() {
+
+        beforeEach(function() {
+          this.cwd = tmp.dirSync();
+          this.cargoToml = path.join(this.cwd.name, 'Cargo.toml');
+
+          fs.writeFileSync(this.cargoToml, [
+            '[package]',
+            'name = "foo"',
+            'version = "1.2.3"',
+            '',
+            '[dependencies.bar]',
+            'version = "2.0.0"',
+            ''
+          ].join('\n'));
+        });
+
+        afterEach(function() {
+          fs.unlinkSync(this.cargoToml);
+          this.cwd.removeCallback();
+        });
+
+        it('should be able to preserve the version', function(done) {
+          presets.updateVersion.cargo({}, this.cwd.name, '1.2.3', (error) => {
+            m.chai.expect(error).to.not.exist;
+
+            m.chai.expect(fs.readFileSync(this.cargoToml, 'utf8')).to.equal([
+              '[package]',
+              'name = "foo"',
+              'version = "1.2.3"',
+              '',
+              '[dependencies.bar]',
+              'version = "2.0.0"',
+              ''
+            ].join('\n'));
+
+            done();
+          });
+        });
+
+      });
+
+      describe('Cargo.toml with a non-semver version', function() {
+
+        beforeEach(function() {
+          this.cwd = tmp.dirSync();
+          this.cargoToml = path.join(this.cwd.name, 'Cargo.toml');
+
+          fs.writeFileSync(this.cargoToml, [
+            '[package]',
+            'name = "foo"',
+            'version = "not-a-semver"',
+            ''
+          ].join('\n'));
+        });
+
+        afterEach(function() {
+          fs.unlinkSync(this.cargoToml);
+          this.cwd.removeCallback();
+        });
+
+        it('should be able to correct the version', function(done) {
+          presets.updateVersion.cargo({}, this.cwd.name, '1.1.0', (error) => {
+            m.chai.expect(error).to.not.exist;
+
+            m.chai.expect(fs.readFileSync(this.cargoToml, 'utf8')).to.equal([
+              '[package]',
+              'name = "foo"',
+              'version = "1.1.0"',
+              ''
+            ].join('\n'));
+
+            done();
+          });
+        });
+
+      });
+
+      describe('Cargo.lock with missing version', function() {
+
+        beforeEach(function() {
+          this.cwd = tmp.dirSync();
+          this.cargoToml = path.join(this.cwd.name, 'Cargo.toml');
+          this.cargoLock = path.join(this.cwd.name, 'Cargo.lock');
+
+          fs.writeFileSync(this.cargoToml, [
+            '[package]',
+            'name = "foo"',
+            'version = "1.0.0"',
+            '',
+            '[dependencies.bar]',
+            'version = "2.0.0"',
+            ''
+          ].join('\n'));
```

Comment:
Along similar lines to what you're testing in Cargo.lock here, there should probably also be a separate test with Cargo.toml set to:
```js
fs.writeFileSync(this.cargoToml, [
  '[package]',
  'name = "foo"',
  '',
  '[dependencies.bar]',
  'version = "2.0.0"',
  ''
].join('\n'));
```
?

Reply:
We have a similar test, I will move it next to this one and extend it to include the `dependencies.bar` section.